src/components/MapControls/MapControls.js

```diff
@@ -47,18 +47,24 @@ const MapControls = ({
   fitExtent,
   children,
 }) => {
-  const { t } = useTranslation();
+  const { t, i18n } = useTranslation();
   const dispatch = useDispatch();
   const map = useSelector((state) => state.app.map);
   const displayMenu = useSelector((state) => state.app.displayMenu);
   const [geolocationFeature, setGeolocationFeature] = useState(null);
   const [geolocating, setGeolocating] = useState(false);
   const featureRef = useRef(geolocationFeature);
+  const [, setForceRender] = useState(0);
   const setGeolocFeatureWithRef = (feature) => {
     featureRef.current = feature;
     setGeolocationFeature(feature);
   };
 
+  useEffect(() => {
+    // Ensure the translations are always applied
+    setForceRender((prevcount) => prevcount + 1);
+  }, [i18n.language]);
```

Comment:
If you have to do this there is probably a problem somewhere else . What is the problem you are trying to fix ?

Reply:
When loading railplus topic the mapcontrols fail to rerender after the initial render, even if the language changes in the app state

Reply:
fixed on master